Plugin/LSP/LSPOutlineView.cpp

```diff
@@ -0,0 +1,1160 @@
+
+namespace {
+    class wxIntClientData : public wxClientData {
+        int m_value = 0;
+    public: 
+        wxIntClientData(int value) : wxClientData(), m_value(value) {}
+        ~wxIntClientData() = default;
+        
+        int GetValue() const { return m_value; }
+        void SetValue(int value) { m_value = value; }
+    };
+}
+
+LSPOutlineView::LSPOutlineView(wxWindow* parent, wxWindowID id, long style)
+    : wxPanel(parent, id), m_style(style)
+{
+    // initialize behaviour from config
+    m_initialiseAllCollapsed = clConfig::Get().Read(kConfigOutlineStartAllCollapsed, false);
+    m_initialiseClassesCollapsed = clConfig::Get().Read(kConfigOutlineStartClassesCollapsed, false);
+    m_initialiseSubclassesCollapsed = clConfig::Get().Read(kConfigOutlineStartSubclassesCollapsed, true);
+    m_initialiseEnumsCollapsed = clConfig::Get().Read(kConfigOutlineStartEnumsCollapsed, true);
+    m_initialiseFunctionsCollapsed = clConfig::Get().Read(kConfigOutlineStartFunctionsCollapsed, true);
+    m_initialiseContainersCollapsed = clConfig::Get().Read(kConfigOutlineStartContainersCollapsed, false);
+    m_initialiseVariablesCollapsed = clConfig::Get().Read(kConfigOutlineStartVariablesCollapsed, true);
+    m_keepNamespacesExpanded = clConfig::Get().Read(kConfigOutlineKeepNamespacesExpanded, true);
+    m_showDetails = clConfig::Get().Read(kConfigOutlineShowSymbolDetails, true);
+    m_showSymbolKind = clConfig::Get().Read(kConfigOutlineShowSymbolKinds, false);
+    m_forceTree = clConfig::Get().Read(kConfigOutlineForceTree, true);
+    m_sortType = (LSPSymbolParser::SortType) clConfig::Get().Read(kConfigOutlineSortType, (int)LSPSymbolParser::SortType::SORT_LINE);
+    
+    // load bitmaps
+    auto images = clGetManager()->GetStdIcons();
+    m_bitmaps.reserve((int)LSPSymbolStyle::BITMAP_MAX);    
+    // order has to be same as LSPSymbolStyle::BITMAP enum
+    m_bitmaps.push_back(images->LoadBitmap("symbol-module"));
+    m_bitmaps.push_back(images->LoadBitmap("symbol-enum"));
+    m_bitmaps.push_back(images->LoadBitmap("symbol-enummember"));
+    m_bitmaps.push_back(images->LoadBitmap("symbol-class"));
+    m_bitmaps.push_back(images->LoadBitmap("symbol-struct"));
+    m_bitmaps.push_back(images->LoadBitmap("symbol-function"));
+    m_bitmaps.push_back(images->LoadBitmap("symbol-constructor"));
+    m_bitmaps.push_back(images->LoadBitmap("symbol-variable"));
+    m_bitmaps.push_back(images->LoadBitmap("symbol-container"));
+    m_bitmaps.push_back(images->LoadBitmap("symbol-namespace"));
+    
+    // create controls
+    CreateUI();      
+    // initialize content (empty)
+    DoInitialiseEmpty();    
+
+    EventNotifier::Get()->Bind(wxEVT_LSP_DOCUMENT_SYMBOLS_REQUESTED, &LSPOutlineView::OnSymbolsRequested, this); 
+}
+
+LSPOutlineView::~LSPOutlineView() 
+{    
+    EventNotifier::Get()->Unbind(wxEVT_LSP_DOCUMENT_SYMBOLS_REQUESTED, &LSPOutlineView::OnSymbolsRequested, this); 
+}
+
+void LSPOutlineView::CreateUI()
```

Comment:
CodeLite dialogs and other UI components are created using `wxCrafter`. It makes the UI easier to maintain and separates the UI from the logic by creating a subclass. Please move this hand-crafted class into a wxCrafter file.